include/npcomp/Dialect/TCP/IR/TCPOps.td

```diff
@@ -54,4 +54,22 @@ def TCP_SplattedOp : TCP_Op<"splatted"> {
   let assemblyFormat = "$splatVal `,` $shape attr-dict `:` functional-type(operands, results)";
 }
 
+def TCP_PadOp : TCP_Op<"pad"> {
+  let summary = "Pads a tensor with a fill value";
+  let description = [{
+    Pads a tensor with `fillVal` along the borders of each dimension according to `lowerExpansion` and `upperExpansion`. Note that this op is unmanaged, meaning that it assumes its operands and their shapes are valid.
+
+    The tensors have dimensions:
+    - operand:   [D1, D2, ..., DN]
+    - lowerExpansion: [L1, L2, ..., LN]
+    - upperExpansion: [U1, U2, ..., UN]
+    - fillVal:   scalar
+    - result:    [D1+L1+U1, D2+L2+U2, ..., DN+LN+UN]
+  }];
+  let arguments = (ins AnyRankedTensor:$operand, Shape_ExtentTensorType:$lowerExpansion, Shape_ExtentTensorType:$upperExpansion, AnyType:$fillVal);
```

Comment:
Can you use `Variadic<Index>` instead of Shape_ExtentTensorType?

Reply:
What's the motivation for re-typing padding specs from 1D tensors into dynamic `Index` arguments? I'm sure there is a good reason, but I may skip this for now. For the most part, I'm basing the pad-op's code off of the other TCP ops rather than the new `pad_tensor`. We can always come back and either edit or deprecate the TCP pad op.

Reply:
Yeah, I guess it's consistent with what we already have. I was thinking ahead to aligning this with linalg.pad_tensor (see link below).

Reply:
Thanks Sean. I've read `linalg.pad_tensor`'s review. It seems that there will also be a `memref`'ed version. But, this is out of this PR's scope.